Xamarin.Forms.Core/Picker.cs

```diff
@@ -1,62 +1,333 @@
-			get { return (int)GetValue(SelectedIndexProperty); }
-			set { SetValue(SelectedIndexProperty, value); }
-		}
-
-		public Color TextColor
-		{
-			get { return (Color)GetValue(TextColorProperty); }
-			set { SetValue(TextColorProperty, value); }
-		}
-
-		public string Title
-		{
-			get { return (string)GetValue(TitleProperty); }
-			set { SetValue(TitleProperty, value); }
-		}
-
-		public event EventHandler SelectedIndexChanged;
-
-		static object CoerceSelectedIndex(BindableObject bindable, object value)
-		{
-			var picker = (Picker)bindable;
-			return picker.Items == null ? -1 : ((int)value).Clamp(-1, picker.Items.Count - 1);
-		}
-
-		void OnItemsCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
-		{
-			SelectedIndex = SelectedIndex.Clamp(-1, Items.Count - 1);
-		}
-	}
+    [RenderWith(typeof(_PickerRenderer))]
+    public class Picker : View
+    {
+        public static readonly BindableProperty TextColorProperty =
+            BindableProperty.Create(
+                nameof(TextColor),
+                typeof(Color),
+                typeof(Picker),
+                Color.Default);
+
+        public static readonly BindableProperty TitleProperty =
+            BindableProperty.Create(
+                nameof(Title),
+                typeof(string),
+                typeof(Picker),
+                default(string));
+
+        public static readonly BindableProperty SelectedIndexProperty =
+            BindableProperty.Create(
+                nameof(SelectedIndex),
+                typeof(int),
+                typeof(Picker),
+                -1,
+                BindingMode.TwoWay,
+                propertyChanged: OnSelectedIndexChanged,
+                coerceValue: CoerceSelectedIndex);
+
+        public static readonly BindableProperty SelectedValueMemberPathProperty =
+            BindableProperty.Create(
+                nameof(SelectedValueMemberPath),
+                typeof(string),
+                typeof(Picker));
+
+        public static readonly BindableProperty ItemsSourceProperty =
+            BindableProperty.Create(
+                nameof(ItemsSource),
+                typeof(IList),
+                typeof(Picker),
+                default(IList),
+                propertyChanged: OnItemsSourceChanged);
+
+        public static readonly BindableProperty DisplayMemberPathProperty =
+            BindableProperty.Create(
+                nameof(DisplayMemberPath),
+                typeof(string),
+                typeof(Picker),
+                propertyChanged: OnDisplayMemberPathChanged);
+
+        public static readonly BindableProperty SelectedItemProperty =
+            BindableProperty.Create(
+                nameof(SelectedItem),
+                typeof(object),
+                typeof(Picker),
+                null,
+                BindingMode.TwoWay,
+                propertyChanged: OnSelectedItemChanged);
+
+        static readonly BindablePropertyKey s_selectedValuePropertyKey =
```

Comment:
you already have SelectedIndex and SelectedItem. there's no need for SelectedValue